feat: allow `while(true)` in `no-constant-condition` (#18286)
eslint/eslint: docs/src/rules/no-constant-condition.md

### checkLoops

> This is an object option having following values:

Author: Done!

* `"all"` - Disallow constant expressions in all loops.

> * `"allExceptWhileTrue"` (default) - Disallow constant expressions in all loops except `while` loop with expression `true`.

Author: Done!

> * `all` - Disallow constant expressions in all loops.
> * `allExceptWhileTrue` (default) - Disallow constant expressions in all loops except `While` loop with expression `true`.
> * `none` - Allow constant expressions in loops.
> 
> Or instead you can set the `checkLoops` value to booleans where `true` is same as `all` and `false` is same as `none`.
> 
> Example of **incorrect** code for when `checkLoops` is `all` or `true`:

Reviewer: We typically use double quotes for string values.

```suggestion
* `"all"` - Disallow constant expressions in all loops.
* `"allExceptWhileTrue"` (default) - Disallow constant expressions in all loops except `while` loops with expression `true`.
* `"none"` - Allow constant expressions in loops.

Or instead you can set the `checkLoops` value to booleans where `true` is same as `"all"` and `false` is same as `"none"`.

Examples of **incorrect** code for when `checkLoops` is `"all"` or `true`:
```

Author: Done!

::: incorrect

```js
/*eslint no-constant-condition: ["error", { "checkLoops": "all" }]*/

while (true) {
    doSomething();
};

for (;true;) {
    doSomething();
};
```

:::

::: incorrect

```js
/*eslint no-constant-condition: ["error", { "checkLoops": true }]*/

while (true) {
    doSomething();
};

do {
    doSomething();
} while (true)
```

:::

Examples of **correct** code for when `checkLoops` is `"all"` or `true`:

::: correct

```js
/*eslint no-constant-condition: ["error", { "checkLoops": "all" }]*/

while (a === b) {
    doSomething();
};
```

:::

::: correct

```js
/*eslint no-constant-condition: ["error", { "checkLoops": true }]*/

for (let x = 0; x <= 10; x++) {
    doSomething();
};
```

:::

> Examples of **correct** code for when `checkLoops` is `"allExceptWhileTrue"`:

Reviewer: ```suggestion
Example of **correct** code for when `checkLoops` is `"allExceptWhileTrue"`:
```

Author: Done!

::: correct

```js
/*eslint no-constant-condition: "error"*/

while (true) {
    doSomething();
};
```

:::

> Examples of **correct** code for when `checkLoops` is `none` or `false`:

Reviewer: ```suggestion
Examples of **correct** code for when `checkLoops` is `"none"` or `false`:
```

    }
};

do {
    doSomething();
    if (condition()) {
        break;
    }
} while (true)
```

:::

::: correct

```js

> /*eslint no-constant-condition: ["error", { "checkLoops": "none" }]*/

Reviewer: Did you mean `"checkLoops": false`? Because `"checkLoops": "none"` is used in the previous block.

```suggestion
/*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
```
